spec/index.adoc

```diff
@@ -1674,6 +1674,100 @@ Any user can top-up any canister this way.
 
 This method is only available in local development instances.
 
+
+[#ic-bitcoin-canisters]
+== The IC bitcoin canisters
+
+Similar to the <<ic-management-canister,management canister>>, the Bitcoin functionality is also exposed through a facade, which we refer to as the "Bitcoin canisters".
+There are two Bitcoin canisters differing only in that one operates on the real Bitcoin blockchain whereas the other only works with the Bitcoin testnet.
+The Bitcoin canisters can be accessed using their addresses:
+
+- The canister address of the Bitcoin canister that uses the real Bitcoin blockchain is `u25tu-yqaaa-aaaaa-aaaaa-eai`.
+- The canister address of the Bitcoin canister that uses the Bitcoin testnet is `jf4vc-xabaa-aaaaa-aaaaa-eai`.
```

Comment:
I am starting to reconsider these IDs. These IDs were generated based on Björn's suggestion:

> Currently we use everything with nat64 | 0101 as a typical canister id, and I could imagine using nat64(i) | 0201 for the Bitcoin canisters, with i in {0,1}.

And then I built on that suggestion to generate them as follows:
```
> Principal.fromUint8Array(Uint8Array.from([0,0,0,0,0,0,0,0,2,1])).toText()
'u25tu-yqaaa-aaaaa-aaaaa-eai'
> Principal.fromUint8Array(Uint8Array.from([1,0,0,0,0,0,0,0,2,1])).toText()
'jf4vc-xabaa-aaaaa-aaaaa-eai'
```

The problem I'm starting to see is that these canister IDs don't play nice with our implementation of `RoutingTable`, which assumes that all canisters are nat64 | 0101 (which these are not). This means that we can never use these canister IDs as part of the `RoutingTable` if we wanted to at some point.

I don't have a concrete suggestion yet, but given that some components in our implementation already assume that canisters are always nat64 | 0101, I do think it'd be beneficial for these canisters to also be nat64 | 0101 to avoid surprises.

Reply:
Where does the routing table (at least the publicly visible part of it, e.g. canister_range) assume a certain length? I thought we carefully avoided such implementation details leaking into the interfaces?

Or are you indeed talking about the internal stuff? Then surely the routing table can be changed?

Reply:
Well, yes and no. The Bitcoin canister (facade) _is not a canister_ in the same sense that the IC management canister (facade) is not a canister. If it were a single canister, then it would need to be deployed on a single subnet. But that's not how we intend to implement it, correct?

Reply:
Why is it not part of `aaaaa-aa` again? It doesn't seem to have much different characteristics than other methods there - like `raw_rand` it's not just canister management, and like most methods it requires routing to another subnet.

Reply:
Ah, because you want to be able to switch networks based on canister IDs? Wouldn't that work better if it's an enum argument? Also easier if more Bitcoin clones need to be supported in the future.

Reply:
In discussion with @akhi3030: Why don't we just add the methods to IC00?

Reply:
Yes, let's just add them to the management canister. The consequences of different canister IDs on the implementation are rather complicated. I'll update this PR.

Reply:
I had a chat with @THLO about this. We introduced separate canister IDs initially because we thought it's more elegant. Callers won't have to send an enum with which network they want to talk to for every request, and we wouldn't have to prefix all these methods with something like `bitcoin_`.

I have an idea of a simple implementation that will allow us to keep those separate canister IDs that are currently specified in the spec. I'll prototype it to see if it works and will get back to you all.